action-allowedlist/approved.json

```diff
@@ -452,6 +462,11 @@
         "actionVersion": "28f608baaa037465aa887c21a7924e9e6b66a9e0",
         "tag": "3.13"
     },
+    {
+        "actionLink": "mxschmitt/action-tmate",
+        "actionVersion": "a283f9441d2d96eb62436dc46d7014f5d357ac22",
+        "tag": "3.17"
+    },
```

Comment:
I need to review this one carefully before approving.

Reply:
I have approved a prior version of `action-tmate`, but I [see something](https://github.com/mxschmitt/action-tmate/blob/master/src/index.js#L104) that I don't like:

```
        const tmateReleaseTar = await tc.downloadTool(`https://github.com/tmate-io/tmate/releases/download/${TMATE_LINUX_VERSION}/tmate-${TMATE_LINUX_VERSION}-static-linux-${tmateArch}.tar.xz`);
```

Thus although we are pinning the action to a hash value, the action itself is downloading a release that is _not_ protected by a hash value. Thus we have no control over what is actually downloaded with the command above.

I denied an action last week for a very similar reason, and there are some other existing actions that I need to review. There is a mitigating factor that I haven't had time to fully think about yet: as I remember it, the main concern is what happens if someone steals GitHub credentials and replaces the binary above with a tainted / malicious version. Now that GitHub _requires_ two factor authentication, this is much less of a concern. So maybe it is ok.

But, I also need to look at the source repository for tmate-io above for review.

Before doing that, @bradbanister @andonyns when was the last time we actually used tmate to SSH into the Docker agent? Do we really need this?

Reply:
I use tmate whenever there is a error in a GitHub action and I don't understand what is going on. Typically it stems from some unexpected difference between the GitHub action environment and my dev environment. The latest example was a  test with a sort ordering difference between Windows and Linux (uppercase vs lowercase characters coming first). Without tmate it would have been a bunch of trial-and-error commits and pushes to diagnose and fix.

Reply:
I think the action is very useful but I haven't actually used it before, when I've thought about using it, is for an scenario where it didn't apply (I can't remember the scenarios right now)